src/NUnitFramework/tests/Constraints/EndsWithConstraintTests.cs

```diff
@@ -36,6 +38,102 @@ public void RespectsCulture()
+        [TestCase(" SS", "s", StringComparison.InvariantCulture)]
+        [TestCase(" ss", "ß", StringComparison.InvariantCultureIgnoreCase)]
+        [TestCase(" ss", "s", StringComparison.InvariantCultureIgnoreCase)]
+        [TestCase(" SS", "s", StringComparison.InvariantCultureIgnoreCase)]
+        [TestCase(" ss", "ß", StringComparison.Ordinal)]
+        [TestCase(" ss", "s", StringComparison.Ordinal)]
+        [TestCase(" SS", "s", StringComparison.Ordinal)]
+        [TestCase(" ss", "ß", StringComparison.OrdinalIgnoreCase)]
+        [TestCase(" ss", "s", StringComparison.OrdinalIgnoreCase)]
+        [TestCase(" SS", "s", StringComparison.OrdinalIgnoreCase)]
+        public void SpecifyComparisonType(string actual, string expected, StringComparison comparison)
+        {
+            // Get platform-specific StringComparison behavior
+            var shouldSucceed = actual.EndsWith(expected, comparison);
+
+            Constraint constraint = Does.EndWith(expected).Using(comparison);
+            if (!shouldSucceed)
+                constraint = new NotConstraint(constraint);
+
+            Assert.That(actual, constraint);
+        }
+
+        [TestCase(" ss", "ß")]
+        [TestCase(" ß", "ß")]
+        [TestCase(" ss", "s")]
+        [TestCase(" SS", "s")]
+        public void SpecifyCultureInfo(string actual, string expected)
+        {
+            var cultureInfo = new CultureInfo("de-DE");
+            // Get platform-specific StringComparison behavior
+            var shouldSucceed = actual.EndsWith(expected, false, cultureInfo);
+
+            Constraint constraint = Does.EndWith(expected).Using(cultureInfo);
+            if (!shouldSucceed)
+                constraint = new NotConstraint(constraint);
+
+            Assert.That(actual, constraint);
+        }
+
+        [Test]
+        public void UseDifferentComparisonTypes_ThrowsException()
+        {
+            var endsWithConstraint = (EndsWithConstraint)TheConstraint;
+
+            Assert.Multiple(() =>
+            {
+                // Invoke Using method before IgnoreCase
+                Assert.That(() => endsWithConstraint.Using(StringComparison.CurrentCulture).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => endsWithConstraint.Using(StringComparison.InvariantCulture).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => endsWithConstraint.Using(StringComparison.InvariantCultureIgnoreCase).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => endsWithConstraint.Using(StringComparison.Ordinal).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => endsWithConstraint.Using(StringComparison.OrdinalIgnoreCase).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
```

Comment:
I wonder if these should throw.
Yes, they could specify `StringComparison.CurrentCultureIgnoreCase` but this could be the same effect.